test/tests/api/v2_0/os_install_tests.py

```diff
@@ -46,8 +50,75 @@ def teardown(self):
+        current_time = date.datetime.now()
+        while True:
+            if (current_time - start_time).total_seconds() > DEFAULT_TIMEOUT_SEC:
+                raise Exception('Timed out after {0} seconds'.format(DEFAULT_TIMEOUT_SEC))
+                break
+            Api().workflows_get()
+            workflows = loads(self.__client.last_response.data)
+            for w in workflows:
+                # LOG.info('print w : {0}'.format(w))
+                if (w.get('node') == id and w.get('injectableName') == graph_name and
+                        w.get('instanceId') == graph_instance):
+                    status = w.get('status')
+                    # LOG.info('{0} - target: {1}, status: {2}'.format(w.get('injectableName'), id, status))
+                    if status == 'succeeded' or status == 'failed' or status == 'canceled':
+                        LOG.info('{0} - target: {1}, status: {2}'.format(w.get('injectableName'), id, status))
+                        if status == 'failed' or status == 'canceled':
+                            LOG.Warning(' workflow {0}: {1}').format(status, w)
+                        assert_equal(status, 'succeeded',
+                            message='{0} - target: {1}, status: {2}'.format(w.get('injectableName'), id, status))
+                        return
+            time.sleep(10)
+            current_time = date.datetime.now()
+            # LOG.info('current time {0} vs start time {1}'.format(current_time, start_time))
+
     def __post_workflow(self, graph_name, nodes, body):
-        workflows().post_workflows(graph_name, timeout_sec=DEFAULT_TIMEOUT_SEC, nodes=nodes, data=body)
+        # check if NODE_INDEX is set
+        index = None
+        try:
+            index = int(os.environ['NODE_INDEX'])
+        except:
+            LOG.info('NODE_INDEX env is not set')
+            workflows().post_workflows(graph_name, timeout_sec=DEFAULT_TIMEOUT_SEC, nodes=nodes, data=body)
+            return
+
+        # check if index is in the array range
+        nodes = self.__get_compute_nodes()
+        if index >= len(nodes):
+            raise Exception('index is outside the array range index: {0} vs nodes len {1}'.format(index, len(nodes)))
+            return
+        LOG.info('node index is set to {0}'.format(index))
+        node = nodes[index]
+        id = node.get('id')
+        # delete active workflow on the selected node
+        try:
+            Api().nodes_workflow_action_by_id(id, {'command': 'cancel'})
+        except ApiException as e:
+            assert_equal(404, e.status, message='status should be 404')
+        Api().nodes_post_workflow_by_id(id, name=graph_name, body=body)
+        log_context = self.__get_data().get('logContext')
+        if log_context is None:
+            raise Exception('Could not find logContext in {0}'.format(self.__get_data()))
+            return
+        # load graph instance id
+        graph_instance = log_context.get('graphInstance')
```

Comment:
Why not directly use the `instanceId` but rather than turn to use `logContext.graphInstance`? the `instanceId` is defined in graph object model, but logContext is a dynamic property.